### PR TITLE
Improve match exactly error handling

### LINKED ISSUE #4039: Improve error handling for Match Exactly
Changes we should make:

- [x] When there is a `SSLException` user should see notification with text similar to "Collect can't reach the server at..." but start with "Collect can't connect securely to server..."
- [x] When there is a 404 fetching the form list the user should see the "Collect can't reach the server at..." notification

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4039

#### What has been done to verify that this works as intended?

Addded new tests.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should be low risk but good to check the new error message scenario (in the issue).

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.java [hunk 1]
```diff
@@ -577,7 +577,7 @@ public void formListDownloadingComplete(HashMap<String, ServerFormDetails> formL
         } else {
             switch (exception.getType()) {
                 case FETCH_ERROR:
-                case UNKNOWN_HOST:
+                case UNREACHABLE:
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.java:576]: Really we're using this to flag when we can't get to the server so the naming felt like it should be more general.

File: collect_app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -704,4 +704,5 @@
     <string name="form_update_error">Form update failed</string>
     <string name="report_to_project_lead">If you keep having this problem, report it to the person who asked you to collect data.</string>
     <string name="unknown_host_error">Collect can\'t reach the server at %s. Did you enter the URL correctly?</string>
```
Comment (lognaturel) [inline collect_app/src/main/res/values/strings.xml]: Would you rename this resource too, please?

**Review by lognaturel — APPROVED** [review]

**Comment by kkrawczyk123** [comment]
@seadowg I was trying to break my local aggregate but I couldn't get anything more than 404, there are my results:
| PR | 1.27.3       |
| ---- | ----------- |
| Screenshot_20200904-153435_PR | Screenshot_20200904-153440_1273 | 

Do you have any advice on further testing?

**Comment by seadowg** [comment]
@kkrawczyk123 sorry this PR (and the issue) are only concerned with the Match Exactly errors. The Get Blank Form errors are good to check but there was no focus on improving them here.

**Comment by kkrawczyk123** [comment]
@seadowg yeah I know and it's the same for match exactly as for get blank form on PR (at least the beginning of it), but it's in notification and PR is probably behind the master and there is no way to expand the whole text of the error on notification so this one is much more transparent, you can see all text so I attached that one.

**Comment by seadowg** [comment]
@kkrawczyk123 ah ok. Looks good then!

**Comment by seadowg** [comment]
@kkrawczyk123 is this all good then or was there a problem I've not understood?

**Comment by kkrawczyk123** [comment]
@seadowg sorry for the delay I didn't have time to get back on that one I will double-check one more time right now. The only thing that I can think of is that I have no idea how to obtain that error:
` When there is a SSLException user should see notification with text similar to "Collect can't reach the server at..." but start with "Collect can't connect securely to server..."`

**Comment by lognaturel** [comment]
@kkrawczyk123 That's going to be a tricky one to do. The easiest would probably be to set a self-signed SSL certificate for either Central or Aggregate but that is still pretty involved - https://docs.getodk.org/central-install-digital-ocean/#using-a-custom-ssl-certificate. I think we can let the tests handle that rare case and not worry about it.

Also, @kkrawczyk123, @grzesiek2010 and I discussed earlier and agreed to merge this and #4064 and verify them on master. That means the same error state can be verified with both match exactly and manual and hit the latest code. Also, verifying one set of changes acts as a regression check on the other.

**Comment by kkrawczyk123** [comment]
My results:
- not existing local server: "If you keep having this problem, report it to the person who asked you to collect data". 
- invalid url: "Collect can't reach the server at ... Did you enter the URL correctly? If you keep having this problem, report it to the person who asked you to collect data"
- broken local aggregate: "Collect can't reach the server at ... Did you enter the URL correctly? If you keep having this problem, report it to the person who asked you to collect data"
- invalid Central url: "If you keep having this problem, report it to the person who asked you to collect data". 

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Comment by lognaturel** [comment]
Thank you, @kkrawczyk123! I have some more exploration to do around self-issued certificates anyway. If that leads me to set it up, I'll be sure to verify the Collect errors as well.

**Comment by seadowg** [comment]
@kkrawczyk123 ah sorry yeah that's a harder one to reproduce. Hopefully @lognaturel is able to!

**Comment by kkrawczyk123** [comment]
@seadowg I've just done it xd I have used https except for http for local server.

**Comment by lognaturel** [comment]
Ahh, that's a good one, @kkrawczyk123. There are a few different ways to get that message and that's an easy one I hadn't considered. 👍

**Pull request merged**